File: Tree.cpp
#include<bits/stdc++.h>
using namespace std;

class node{
   public:
   string desc;
   node *left, *right, *middle, *menu, *last;
   node(string d){
      desc = d;
      left = NULL, right = NULL, middle = NULL, last = NULL;
   }
};

void dfs(node *root){
    if(root == NULL) return;
    dfs(root->left);
    dfs(root->middle);
    dfs(root->right);
    dfs(root->last);
    cout << root->desc<<"\n";
    return;
}

int main(){
   string  desc = "1. Create new Store 2. Check existing Store";
   node *root = new node(desc);
   desc = "Enter Store Name";
   node* lvl_one_one = new node(desc);
   root->left = lvl_one_one;
   desc  = "Ok, you can use the menu to create, edit and share your shop 1. Add product2. manage inventory3. website link 4. Exit";
   node *lvl_two_one = new node(desc);
   lvl_one_one->right = lvl_two_one;
   
   // add product tree 
   
   desc = "Ok what is your product name ?";
   node *lvl_three_one = new node(desc);
   lvl_two_one->left = lvl_three_one;
   lvl_three_one->menu = lvl_two_one;
   desc = "what is your product  description ?";
   node *lvl_four_one = new node(desc);
   lvl_three_one->left = lvl_four_one;
   lvl_four_one->menu = lvl_two_one;
   desc = "Can you add your product image ?";
   node *lvl_five_one = new node(desc);
   lvl_four_one->left = lvl_five_one;
   lvl_five_one->menu = lvl_two_one;
   desc = " Ok what is your product's inventory ?";
   node *lvl_six_one = new node(desc);
   lvl_five_one->left = lvl_six_one;
   lvl_six_one->menu = lvl_two_one;

  // update inventory tree

   desc = "Write your product name below";
   node *lvl_three_two = new node(desc);
   lvl_two_one->middle = lvl_three_two;
   lvl_three_two->menu = lvl_two_one;
   desc = "What is your product's current inventory ?";
   node *lvl_four_two = new node(desc);
   lvl_three_two->left = lvl_four_two;
   lvl_four_two->menu = lvl_two_one;

   // website link and exit tree 
   desc = "website link";
   node *lvl_three_three = new node(desc);
   lvl_two_one->right = lvl_three_three;
   lvl_three_three->menu = lvl_two_one;
   desc = "have a nyc day :) ";
   node *lvl_three_four = new node(desc);
   lvl_two_one->last = lvl_three_four;
   lvl_three_four->menu = lvl_two_one;

   // existing store logic 
   desc = "Enter your existing Store name ";
   node *lvl_one_two = new node(desc);
   root->right = lvl_one_two;
   lvl_one_two->left = lvl_two_one;

   dfs(root);
    return 0;
}

File: Tree.js

class node{
   constructor(desc,state,type){
      this.desc = desc
      this.state = state
      this.type = type
	  this.left = null, this.right = null, this.middle = null;
	  this.last = null, this.menu = null;
   }
}


   let desc = ` ____WELCOME_____
         1. Create new Store 
         2. Check existing Store
               `
   let root = new node(desc,"choose","int");
   desc = "Enter Store Name"
   let lvl11 = new node(desc,"pick","new-store");

  root.left = lvl11;
  desc = `Now, you can use the menu to create, edit and share your shop  
       1. Add product
       2. manage inventory
       3. website link 
       4. Exit
        `
  let lvl21 = new node(desc,"choose","int");
  lvl11.left = lvl21;
   
  // add product tree 
   
  desc = "Ok what is your product name ?";
  var lvl31 = new node(desc,"pick","newProduct");
  lvl21.left = lvl31;
  lvl31.menu = lvl21;
  desc = "what is your product  description ?";
  var lvl41 = new node(desc,"pick","string");
  lvl31.left = lvl41;
  lvl41.menu = lvl21;
  desc = "Can you add your product image ? If not type 'no'";
  var lvl51 = new node(desc,"pick","img");
  lvl41.left = lvl51;
  lvl51.menu = lvl21;
  desc = "Ok what is your product's inventory ?";
  var lvl61  = new node(desc,"pick","int");
  lvl51.left = lvl61;
  lvl61.menu = lvl21;
  desc = "Added !!!";
  var lvl71  = new node(desc,"added","transition");
  lvl61.left = lvl71;
  lvl71.menu = lvl21;

  // update inventory tree

  desc = "Write your product name below";
  var lvl32 = new node(desc,"pick","productCheck");
  lvl21.middle = lvl32;
  lvl32.menu = lvl21;
  desc = "What is your product's current inventory ?";
  var lvl42 = new node(desc,"pick","int");
  lvl32.left = lvl42;
  lvl42.menu = lvl21;
   desc = "updated !!!";
  var lvl52 = new node(desc,"updated","transition");
  lvl42.left = lvl52;
  lvl52.menu = lvl21;

  // website link and exit tree 
  desc = "website link";
  var lvl33 = new node(desc,"link","transition");
  lvl21.right = lvl33;
  lvl33.menu = lvl21;
  desc = "have a nyc day ::) ";
  var lvl34 = new node(desc,"exit","exit");
  lvl21.last = lvl34;


  // existing store logic 
  desc = "Enter your existing Store name ";
  var lvl12 = new node(desc,"pick","store-existing");
  root.middle = lvl12;
  lvl12.left = lvl21;
  
  root.menu = lvl21

//   dfs(root);

module.exports = {
   ...root
}


// one lvl below export root 
